src/factors/Bearing2D.jl

```diff
@@ -40,12 +37,11 @@ function (cfo::CalcFactor{<:Pose2Point2Bearing})( res::AbstractVector{<:Real},
   reuse.resid .-= reuse.predvec
 
   # must return result of length zDim==1 in this case
-  res[1] = sum(abs.(reuse.resid))
+  return sum(abs.(reuse.resid))
```

Comment:
we will probably drop the reuse.residual container here at some point, right?

Reply:
Yes, there are a couple of different ways in use currently. I agree we can experiment and then standardise.